torch/distributed/_tensor/device_mesh.py

```diff
@@ -162,11 +165,16 @@ def _get_or_create_default_group(self):
                 # automatically set the current cuda device base on num of gpu devices available in each host
                 # NOTE: This device selection would only work for homogeneous hardware.
                 torch.cuda.set_device(get_rank() % torch.cuda.device_count())
-            # TODO (xilunwu): to perform DTensor random ops, we need to ensure all ranks in mesh is initialized
-            # with the same random seed. The seed to use will be the current seed on rank 0. We store this seed
-            # as an attribute of device mesh for future use. However, the detail is still TBD how we gonna use
-            # this attribute, so we will implement this logic once we figure out the answer.
+            # To perform DTensor random ops, we need to ensure all ranks in mesh is initialized
+            # with the same random seed. The seed to use will be a random number sampled on rank 0.
+            # We store this seed as an attribute of device mesh for DTensor's random ops use only.
             self._seed = torch.cuda.initial_seed()
+            random.seed(self._seed)
+            obj_list = [random.randint(0, sys.maxsize)]
+            broadcast_object_list(obj_list)
```

Comment:
I think we probably want to first turn on this seed sync in `parallelize_module` api instead of doing it in DeviceMesh here, we should study the UX before turning it on by default in device_mesh.py.

And also, this seed_sync should be inside a RNGStateTracker class `init` instead, I feel we spawn random ops stuff here and there, and we should make it more organized.